lib/authentication/signup/models/email.dart

```diff
@@ -0,0 +1,30 @@
+import 'package:formz/formz.dart';
+
+enum EmailValidationError { empty, invalidEmail }
+
+class Email extends FormzInput<String, EmailValidationError> {
+  const Email.pure() : super.pure('');
+  const Email.dirty([super.value = '']) : super.dirty();
+
+  get invalid => null;
+
+  @override
+  EmailValidationError? validator(String value) {
+    if (value.isEmpty) {
+      return EmailValidationError.empty;
+    }
+    final emailRegex = RegExp(r'^[^@]+@[^@]+\.[^@]+$');
+    return emailRegex.hasMatch(value)
+        ? null
+        : EmailValidationError.invalidEmail;
+  }
+
+  String validationError(EmailValidationError error) {
```

Comment:
lets keep this in the widget instead pls. The widget that uses this state should be deciding how to display the error message. We shouldn't be defining that here